app/models/bulkrax/xml_etd_dc_entry.rb

```diff
@@ -0,0 +1,113 @@
+      self.parsed_metadata[work_identifier] = [self.raw_metadata[source_identifier]]
+      xml_elements.each do |element_name|
+        next if complicated_elements.include?(element_name)
+        elements = record.xpath("//*[name()='#{element_name}']")
+        next if elements.blank?
+        elements.each do |el|
+          el.children.map(&:content).each do |content|
+            add_metadata(element_name, content) if content.present?
+          end
+        end
+      end
+      add_model
+      add_visibility
+      add_rights_statement
+      add_admin_set_id
+      add_collections
+      add_authoridentifier
+      add_subject
+      add_identifier
+      add_language
+      # alt identifier
+      add_creator
+      add_contributor
+      # etc
+
+      self.parsed_metadata['file'] = self.raw_metadata['file']
+
+      add_local
+      raise StandardError, "title is required" if self.parsed_metadata['title'].blank?
+      self.parsed_metadata
+    end
+
+    def add_model
+      self.parsed_metadata['model'] = 'ThesisOrDissertation'
+    end
+
+    def add_authoridentifier
+      add_complicated_element('authoridentifier_isni', 'authoridentifier', 'uketdterms:ISNI')
+      add_complicated_element('authoridentifier_orcid', 'authoridentifier', 'uketdterms:ORCID')
+    end
+
+    def add_subject
+      add_complicated_element('subject', 'subject', 'dcterms:Ddc')
+    end
+
+    def add_identifier
+      add_complicated_element('identifier', 'identifier', 'dcterms:DOI')
+    end
+
+    def add_language
+      add_complicated_element('language', 'language', 'dcterms:ISO639-2')
+    end
+
+    def add_complicated_element(element_label, element_name, type_value)
+      elements = record.xpath("//*[name()='#{element_name}']")
+      return if elements.blank?
+      elements.each do |el|
+        el.children.map(&:content).each do |content|
+          add_metadata(element_label, content) if content.present? && el.attr('type') == type_value
+        end
+      end
+    end
+
+    def add_creator
+      add_name_field('creator', 'creator')
+    end
+
+    def add_contributor
+      add_name_field('contributor', 'advisor')
+    end
+
+    def add_name_field(name_field_prefix, element_name)
+      elements = record.xpath("//*[name()='#{element_name}']")
+      return if elements.blank?
+      position = 0
+      elements.each do |el|
+        el.children.map(&:content).each do |content|
+          names = content.split(/\s*;\s*/)
+          next if names.blank?
+          names.each do |name|
+            separated_name = name.split(/\s*,\s*/)
+            next if separated_name.blank?
+            add_metadata("#{name_field_prefix}_family_name", (separated_name.first || ''))
+            add_metadata("#{name_field_prefix}_given_name", (separated_name.length > 1 ? separated_name.last : ''))
+            add_metadata("#{name_field_prefix}_name_type", 'Personal')
+            add_metadata("#{name_field_prefix}_position", position)
```

Comment:
I see you declared `position = 0` then reference it here.  Should we be incrementing the position value between lines 98 and 99?

Reply:
This was a mistake on my part. I added it but must have deleted the increment at some point accidentally.